DANS-KNAW/easy-ingest-dispatcher
# Pid microservice integration

fixes EASY-1115

#### When applied it will
* do the integration with the hazelcast pid microservice

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?
1. checkout and deploy this and all related pull requests as listed below
  a. **NOTE:** to let the pid-generator work with Hazelcast, please change `easy_pid_generator_mode` in easy-dtap/all.yml to `hazelcast` rather than `rest`.
1. verify that the services Hazelcast, Pid-Generator and Ingest-Dispatcher are all up and running. Use `sudo service <service-name> status` to do so.
1. check the log/err files of these services in `/var/log/<service-name>`. They should be connected to the Hazelcast network and should be waiting for new jobs to come in.
1. to see the next steps in action, log in to deasy with a separate terminal and run `tail -f` on the log file of the Ingest-Dispatcher service.
1. do a deposit, either using Sword or by copy-pasting a deposit in the folder identified in the log of the previous step. make sure the folder has 'group-write-access'.
1. once the deposit is in this folder, the Ingest-Dispatcher will pick this up within 5 seconds and start consuming the deposit. the log in the separate terminal should give an idea of what the application is doing.
1. once the depositing process is done, the `bag` folder inside `deposit` should have been deleted and the `deposit.properties` should contain a status of `ARCHIVED` and a link to the dataset in the description.
1. the dataset should also be visible in the webui.

#### related pull requests on github
* https://github.com/DANS-KNAW/easy-ingest-flow/pull/54
* https://github.com/DANS-KNAW/easy-dtap/pull/44
* https://github.com/DANS-KNAW/easy-license-creator/pull/10 ***--> merged***
* https://github.com/DANS-KNAW/easy-pid-generator/pull/10
